xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -1977,19 +1980,37 @@ void COutput::InitCycle()
     method = VS_INTERLACEMETHOD_NONE;
     if (m_pp && !m_pp->UpdateDeintMethod(method))
     {
+      CLog::Log(LOGDEBUG, LOGVIDEO, "VAAPI output: Current postproc does not want new deinterlace mode, removing");
       delete m_pp;
       m_pp = NULL;
       DropVppProcessedPictures();
     }
     if (!m_pp)
     {
+      const bool preferVaapiRender = CServiceBroker::GetSettings().GetBool(CSettings::SETTING_VIDEOPLAYER_PREFERVAAPIRENDER);
+      // For 1080p/i or below, always use CVppPostproc even when not deinterlacing
+      // Reason is: mesa cannot dynamically switch surfaces between use for VAAPI post-processing
+      // and use for direct export, so we run into trouble if we or the user want to switch
+      // deinterlacing on/off mid-stream.
+      // See also: https://bugs.freedesktop.org/show_bug.cgi?id=105145
+      const bool alwaysInsertVpp = (m_config.vidWidth * m_config.vidHeight) <= (1920 * 1080);
```

Comment:
we don't do this on Intel

Reply:
Well then I have to make it a driver-specific quirk. Is that OK?

Reply:
yes, check on open and store in config for later use